example/main.go

```diff
@@ -2,152 +2,262 @@ package main
+type ClaudeResponse struct {
+	Content    []map[string]interface{} `json:"content"`
+	StopReason string                   `json:"stop_reason"`
+	Usage      map[string]int           `json:"usage"`
+}
 
-	// Create API requests
-	openaiRequest, err := openai.NewRequest("https://api.openai.com/v1/chat/completions", cfg.OpenAIAPIKey)
+func main() {
+	// Load .env file
+	err := godotenv.Load(".env")
 	if err != nil {
-		log.Fatalf("Failed to create openai request: %v", err)
+		log.Printf("Warning: Error loading .env file: %v", err)
 	}
-	azureRequest, err := azure.NewRequest(cfg.AzureEndpoint, cfg.AzureAPIKey)
-	if err != nil {
-		log.Fatalf("Failed to create azure request: %v", err)
+
+	// Load configuration from environment variables
+	bedrockConfig := BedrockConfig{
+		AccessKeyID:     os.Getenv("AWS_ACCESS_KEY_ID"),
+		SecretAccessKey: os.Getenv("AWS_SECRET_ACCESS_KEY"),
+		Region:          os.Getenv("AWS_REGION"),
 	}
-	vertexRequest, err := vertex.NewRequest(cfg.VertexProjectID, cfg.VertexLocation)
-	if err != nil {
-		log.Fatalf("Failed to create vertex request: %v", err)
+
+	// Check if Bedrock credentials are available
+	if bedrockConfig.AccessKeyID == "" || bedrockConfig.SecretAccessKey == "" || bedrockConfig.Region == "" {
+		log.Fatalf("AWS Bedrock credentials not found in environment variables")
 	}
 
-	// Get model configuration
-	modelConfig := config.GetModelConfig()
-	modelConfig.Clients = []http.Request{
-		*openaiRequest,
-		*azureRequest,
-		*vertexRequest,
+	fmt.Println("🔑 AWS Bedrock credentials loaded successfully")
+	fmt.Printf("Using AWS region: %s\n", bedrockConfig.Region)
+
+	// Test Bedrock API with Claude model
+	modelID := "anthropic.claude-3-sonnet-20240229-v1:0"
+
+	// Create request payload for Claude
+	request := ClaudeRequest{
+		AnthropicVersion: "bedrock-2023-05-31",
+		MaxTokens:        1000,
+		Messages: []map[string]interface{}{
+			{
+				"role":    "user",
+				"content": "Hello, how are you? Tell me a short joke.",
+			},
+		},
 	}
-	modelConfig.RedisConfig = &cfg.RedisConfig
 
-	// Set up Azure regions
-	modelConfig.AzureRegions = map[string]string{
-		"eastus":     cfg.AzureEndpoint,
-		"westeurope": "https://custom-westeurope.openai.azure.com",
+	// Make request to Bedrock API
+	response, err := invokeBedrockModel(bedrockConfig, modelID, request)
+	if err != nil {
+		log.Fatalf("Failed to invoke Bedrock model: %v", err)
 	}
 
-	// Create transport with configuration
-	transport, err := transport.NewTransport(modelConfig)
+	// Print response
+	fmt.Println("\n🤖 AWS Bedrock Response:")
+	fmt.Println(response)
+}
+
+// invokeBedrockModel makes a request to the AWS Bedrock API
+func invokeBedrockModel(config BedrockConfig, modelID string, request interface{}) (string, error) {
+	// Convert request to JSON
+	jsonData, err := json.Marshal(request)
 	if err != nil {
-		log.Fatalf("Failed to create transport: %v", err)
+		return "", fmt.Errorf("failed to marshal request: %v", err)
```

Comment:
Rule: whenever you're wrapping an error using the `fmt` package, use `%w` (https://pkg.go.dev/fmt#Errorf). It makes it so the error is _actually_ wrapped and can be unwrapped, rather than just squished in as a string.
```suggestion
		return "", fmt.Errorf("failed to marshal request: %w", err)
```